CMakeLists.txt

```diff
@@ -4,18 +4,51 @@ project(pmemkv)
 set(CMAKE_CXX_STANDARD 11)
 set(SOURCE_FILES src/pmemkv.cc src/pmemkv.h)
 set(3RDPARTY ${PROJECT_SOURCE_DIR}/3rdparty)
-include_directories(${3RDPARTY}/gtest)
-include_directories(${3RDPARTY}/gtest/include)
-include_directories(${3RDPARTY}/nvml/include)
+set(GTEST_VERSION 1.7.0)
+
+find_package(PkgConfig QUIET)
+include(ExternalProject)
+
+if(EXISTS ${CMAKE_SOURCE_DIR}/googletest-${GTEST_VERSION}.zip)
+set(GTEST_URL ${CMAKE_SOURCE_DIR}/googletest-${GTEST_VERSION}.zip)
+else()
+set(GTEST_URL https://github.com/google/googletest/archive/release-${GTEST_VERSION}.zip)
+endif()
+ExternalProject_Add(
+	gtest
+	URL ${GTEST_URL}
+	DOWNLOAD_NAME googletest-${GTEST_VERSION}.zip
+	DOWNLOAD_DIR ${CMAKE_SOURCE_DIR}
+	PREFIX ${CMAKE_CURRENT_BINARY_DIR}/gtest
+	INSTALL_COMMAND ""
+	CMAKE_ARGS -DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER}
+)
+ExternalProject_Get_Property(gtest source_dir binary_dir)
+add_library(libgtest IMPORTED STATIC GLOBAL)
+add_dependencies(libgtest gtest)
+set_target_properties(libgtest PROPERTIES
+	"IMPORTED_LOCATION" "${binary_dir}/libgtest.a"
+	"IMPORTED_LINK_INTERFACE_LIBRARIES" "${CMAKE_THREAD_LIBS_INIT}"
+)
+include_directories("${source_dir}/include")
+
+if(PKG_CONFIG_FOUND)
+	pkg_check_modules(PMEMOBJ++ REQUIRED libpmemobj++)
+else()
+	find_package(PMEMOBJ++ REQUIRED)
+endif()
+
+include_directories(${PMEMOBJ++_INCLUDE_DIRS})
+link_directories(${PMEMOBJ++_LIBRARY_DIRS})
 
 add_library(pmemkv ${SOURCE_FILES})
```

Comment:
Taken from the docs:
> If no type is given explicitly the type is STATIC or SHARED based on whether the current value of the variable BUILD_SHARED_LIBS is ON.

So I think it's fine. Let the user decide if it's not a problem.